packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationModule.m

```diff
@@ -0,0 +1,244 @@
+// Copyright 2019-present 650 Industries. All rights reserved.
+
+#import <EXScreenOrientation/EXScreenOrientationModule.h>
+#import <UMCore/UMEventEmitterService.h>
+#import <EXScreenOrientation/EXScreenOrientationRegistry.h>
+#import <UMCore/UMModuleRegistryProvider.h>
+#import <EXScreenOrientation/EXScreenOrientationUtilities.h>
+
+#import <UIKit/UIKit.h>
+
+@interface EXScreenOrientationModule ()
+
+@property (nonatomic, weak) UMModuleRegistry *moduleRegistry;
+@property (nonatomic, weak) id<UMEventEmitterService> eventEmitter;
+@property (nonatomic) bool hasListeners;
+@property (nonatomic) UIInterfaceOrientation currentScreenOrientation;
+
+@end
+
+static int INVALID_MASK = 0;
+static NSString *defaultAppId = @"bareAppId";
+
+@implementation EXScreenOrientationModule
+
+UM_EXPORT_MODULE(ExpoScreenOrientation);
+
+- (instancetype)init
+{
+  if ((self = [super init])) {
+    [[UIDevice currentDevice] beginGeneratingDeviceOrientationNotifications];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(handleDeviceOrientationChange:)
+                                                 name:UIDeviceOrientationDidChangeNotification
+                                               object:[UIDevice currentDevice]];
+  }
+  return self;
+}
+
+- (void)dealloc
+{
+  [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
+{
+  _moduleRegistry = moduleRegistry;
+  _eventEmitter = [moduleRegistry getModuleImplementingProtocol:@protocol(UMEventEmitterService)];
+  _hasListeners = NO;
+  UIDevice *device = [UIDevice currentDevice];
+  UIInterfaceOrientation currentDeviceOrientation = [EXScreenOrientationUtilities UIDeviceOrientationToUIInterfaceOrientation:[device orientation]];
+  UIInterfaceOrientationMask orientationMask = [self orientationMask];
+  
+  // this gives the correct information of screen orientation before any rotation or locking
+  if ([EXScreenOrientationUtilities doesOrientationMask:orientationMask containOrientation:currentDeviceOrientation]) {
+    _currentScreenOrientation = currentDeviceOrientation;
+  } else {
+    _currentScreenOrientation = [EXScreenOrientationUtilities defaultOrientationForOrientationMask:[self orientationMask]];
```

Comment:
What is the meaning of `_currentScreenOrientation`? Is this the desired orientation, or the current one? If it's the current one, why aren't we setting it to the actual current orientation? (I don't quite get the comment over the `if` 😞)

Reply:
It's the current one. Because firstly we check if the current device orientation is compatible with our saved mask. If it true then the screen has the same orientation as the device. Otherwise, we calculate the screen orientation from the saved mask.